src/pages/index.js

```diff
@@ -58,41 +148,66 @@ addCardButton.addEventListener("click", () => {
   newCardPopup.open();
 });
 
-// Initialize an instance
-newCardPopup.setEventListeners();
-
-// UserInfo popup
-// Comment to the reviewer - I changed here the logics to use
-// the PopupWithFormClass for Profile Data change
-// and removed the extension of the popup class to the UserInfo class
-
+// UserInfo Popup
 const userPopupForm = new PopupWithForm({
   popupSelector: "#profileChange",
   handleFormSubmit: (userInput) => {
-    const newUserInput = userInput;
-    userInfo.setUserInfo({
-      name: newUserInput.name,
-      description: newUserInput.description,
-    });
-    userPopupForm.close();
+    userPopupForm.renderLoading(true);
+    api
+      .editProfileData(userInput)
+      .then((data) => {
+        userInfo.setUserInfo({
+          name: data.name,
+          description: data.about,
+        });
+      })
+      .catch((err) => console.error(err))
+      .finally(() => userPopupForm.renderLoading(false));
   },
 });
 
+// PROFILE INFO CHANGES
+
+// User Info and Avatar Change
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__subtitle",
+  avatarSelector: ".profile__avatar",
+});
+
+//Profile Avatar Change
+const profilePicPopup = new ProfilePicPopup({
+  popupSelector: "#profilePicChange",
+  avatarSelector: ".profile__avatar",
+  handleFormSubmit: (inputValues) => {
+    profilePicPopup.renderLoading(true);
+    api
+      .avatarUpdate(inputValues)
+      .then((data) => {
+        profilePicPopup.changeProfilePic(data);
+        userInfo.setNewAvatar(data);
+      })
+      .catch((err) => console.error(err))
+      .finally(() => profilePicPopup.renderLoading(false));
+  },
+});
+
+profileAvatarContainer.addEventListener("click", () => {
+  profilePicPopup.open();
 });
 
 profileButtonEdit.addEventListener("click", () => {
   formValidators["profile-form"].resetValidation();
   userPopupForm.open();
-  const currentUserData = userInfo.getUserInfo();
-  profileModalInputName.value = currentUserData.name;
-  profileModalInputSubtitle.value = currentUserData.description;
+  api
+    .getUserData()
+    .then((data) => {
+      userPopupForm.setInputValues(data);
+    })
+    .catch((err) => console.error(err));
```

Comment:
You had the correct with `userInfo.getUserInfo()`. Please, don't make requests on opening. It's a waste of resources

```js

  const currentUserData = userInfo.getUserInfo();

// then set the data
```

Reply:
I left the logics for getting the user data and the popup opening for the edit profile button press.
the logics for the  submit, data change, and sending to the server  is in the proflechange popup.

```js
profileButtonEdit.addEventListener("click", () => {
  formValidators["profile-form"].resetValidation();
  userPopupForm.open();
  const currentUserData = userInfo.getUserInfo();
});
```